framework/src/org/apache/cordova/CordovaWebViewClient.java

```diff
@@ -89,6 +97,15 @@ public void setWebView(CordovaWebView view) {
         this.appView = view;
         helper = new CordovaUriHelper(cordova, view);
     }
+    
+    /**
+     * Specifies the HTTP auth handler for handling auth challenge requests.
+     *
+     * @param handler HTTP auth handler.
+     */
+    public void setHttpAuthRequestHandler(HttpAuthRequestHandler handler) {
```

Comment:
Right now, almost all plugin hooks are implemented by adding a method to CordovaPlugin and PluginManager (which loops over all plugins and calls the corresponding CordovaPlugin method).

Calling plugins based on what interfaces they implement is a neat idea, although it'd be a new concept to the project. I'm not actually sure if there are benefits to it or not! Likely we could use that info to improve performance a bit, but perf isn't an issue atm afaik.